### DISCUSSION TITLE
docs: Spanish translations / translation best practices

### DISCUSSION THREAD

**Discussion by jodylecompte** [saadeghi/daisyui#2486] [General]
### Describe the issue

Not a problem/bug report, but looking for guidance on translation expectations / best practice and thought an issue might be a better place to start. I am looking to complete the missing Spanish translations, and a few questions came up: 

**Is there a preference for level of detail vs translating as closely as possible?**
For example, `Modal` is the same in both English and Spanish, but I've also seen it referred to as "diálogo modal." (dialog modal) or "ventana modal" (modal window). In a similar vein, "Dropdown" could be translated as simply "Desplegable", while "Menú desplegable" is more explicit but the Dropdown component appears to be able to be used for any element, not necessarily just a dropdown menu. Wanted to confirm which approach is preferable.

**Besides proper nouns, are there any words or phrases that should not be translated **
For example, I thought at first maybe individual component names were untranslated to match the English CSS class names, but spot checking between the different files it seems that some are translated and others are not. Then reading some of the longer translation strings, I came across stuff like this:

```
"Text Input is a simple input field.": "Text Input es un campo de entrada simple.",
"Text Input": "Text Input"
```
Where both instances of `Text Input` are not translated, but further down other inputs like "Data Input" are translated

My assumption is this is just the translations being incomplete, but I wanted to be sure it wasn't intentional before making any modifications.

### What browsers are you seeing the problem on?

All browsers

**Comment by saadeghi** [comment]
Hey. Thanks for the help.  
I think the best approach is to use words and sentences which developers are most comfortable with. In some languages and with some strings, translating everything won't help and it would be even confusing because for example developers are more comfortable with a English word because it's a technical term and it's common to use the English word for it.

I'm not saying the translations are good. All translation files need improvements of course but I think we shouldn't force the same approach of translation every string to the native language, instead we should use the thing that are easy to understand (even if it's not translated)

So for Spanish translation, if you see some parts are not translated and it would be easier to use if it was translated, your contribution is welcome 💚  
And if you see some technical words are translated to Spanish but they sound weird in Spanish and they would sound better untranslated, you can also reverse those.

Let me know if you have any questions.

**Comment by saadeghi** [comment]
Moving this to discussions.

**Comment by jodylecompte** [comment]
Oh cool, I didn't even realize that Github supported discussions now.

Honestly, I'm not really sure which way is more ideal. I'm reasonably fluent in Spanish from a combination of studying out of interest and traveling/work, but I don't have that "native mindset" of what a native speaker would expect.

My gut feeling is that component names should not be translated, and they should also be referred to in English even in Spanish strings like the example above. The rationale being that it's not just a generic "entrada de texto" (text input) but the unique "Text Input" component from DaisyUI. 

But that said, I'd love to get a native Spanish speaker (or anyone whose first language is not English really) to weigh in on what they prefer to see.
